pkgs/racket-doc/scribblings/raco/test.scrbl

```diff
@@ -378,4 +379,32 @@ the test's output is prefixed with a
 
 line.
 
-
+@section{Logging Test Results}
+@defmodule[raco/testing]
+
+This module provides a general purpose library for tracking test results
+and displaying a summary message. The command @exec{raco test} uses this library
+to display test results. Therefore, any testing framework that wants to integrate
+with @exec{raco test} should also use this library to log test results.
+
+@defproc[(test-log! [result any/c]) void?]{
+ Adds a test result to the running log. If @racket[result] is false,
+ then the test is considered a failure.}
+
+@defproc[(test-report [#:display? display? any/c #f]
+                      [#:exit? exit? any/c #f])
+         (cons/c exact-nonnegative-integer?
+                 exact-nonnegative-integer?)]{
+ Processes the running test log. The first integer is the failed tests, the
+ second is the total tests. If @racket[display?] is true, then a message is
+ displayed. If there were failures, the message is printed on
+ @racket[(current-error-port)]. If @racket[exit?] is true, then if there were
+ failures, calls @racket[(exit 1)].}
+
+@defboolparam[test-log-enabled? enabled? #:value #t]{
+ When set to @racket[#f], @racket[test-log!] is a no-op. This is useful to
+ dynamically disable certain tests whose failures are expected and shouldn't be
+ counted in the test log, such as when testing a custom check's failure
+ behavior.}
+
+@history[#:added "1.13"]
```

Comment:
I’ve just noticed this, but this history note should go under the `defmodule`, right?  It looks extremely out of place since `current-test-invocation-directory` is added, which comes with another history note.

Reply:
That seems reasonable, though I think other modules tend to put them at the end? (Maybe `current-test-invocation-directory`’s note should move?)

Reply:
I agree with @usaoc that this should be placed above, probably immediately before `test-log!`.